<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from gp8_gazebo.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="motoman_gp8">
  <!-- link list -->
  <link name="base_rail">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/slider_barra v1.stl" scale="0.1 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/slider_barra v1.stl" scale="0.1 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="81.012"/>
      <inertia ixx="12.087" ixy="0.0" ixz="0.0" iyy="0.11" iyz="0.002" izz="12.125"/>
    </inertial>
  </link>
  <link name="base_plate">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/slider_base v1.stl" scale="0.1 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/slider_base v1.stl" scale="0.1 0.1 0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="4.362"/>
      <inertia ixx="0.012" ixy="0.0" ixz="0.0" iyy="0.011" iyz="0.0" izz="0.023"/>
    </inertial>
  </link>
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.5490196078431373 0.09019607843137255 0.10196078431372549 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_base_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10.871"/>
      <inertia ixx="0.0587" ixy="0.0" ixz="0.0" iyy="0.0587" iyz="0.0" izz="0.0587"/>
    </inertial>
  </link>
  <link name="link_1_s">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_1_s.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_1_s.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.594"/>
      <inertia ixx="0.16" ixy="0.0" ixz="0.003" iyy="0.2" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <link name="link_2_l">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_2_l.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.5490196078431373 0.09019607843137255 0.10196078431372549 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_2_l.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="9.979"/>
      <inertia ixx="0.168" ixy="0.001" ixz="0.002" iyy="0.117" iyz="0.002" izz="0.117"/>
    </inertial>
  </link>
  <link name="link_3_u">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_3_u.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_3_u.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.862"/>
      <inertia ixx="0.14" ixy="0.0" ixz="0.001" iyy="0.17" iyz="0.0" izz="0.15"/>
    </inertial>
  </link>
  <link name="link_4_r">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_4_r.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.5490196078431373 0.09019607843137255 0.10196078431372549 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_4_r.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.891"/>
      <inertia ixx="0.06" ixy="0.0" ixz="0.0" iyy="0.23" iyz="0.0" izz="0.24"/>
    </inertial>
  </link>
  <link name="link_5_b">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_5_b.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_5_b.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.747"/>
      <inertia ixx="0.2" ixy="0.0" ixz="0.0" iyy="0.2" iyz="0.0" izz="0.2"/>
    </inertial>
  </link>
  <link name="link_6_t">
    <visual>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/visual/gp8_link_6_t.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.5490196078431373 0.09019607843137255 0.10196078431372549 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_gp8_support/meshes/collision/gp8_link_6_t.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.03"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.066" iyz="0.0" izz="0.0065"/>
    </inertial>
  </link>

  <!-- end of link list -->
  <!-- joint list -->
  <joint name="joint_0_s" type="prismatic">
    <parent link="base_rail"/>
    <child link="base_plate"/>
    <origin rpy="0 0 0" xyz="0 0 0.086"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.92" lower="-0.5" upper="0.5" velocity="0.01"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_01_s" type="fixed">
    <parent link="base_plate"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="joint_1_s" type="revolute">
    <parent link="base_link"/>
    <child link="link_1_s"/>
    <origin rpy="0 0 0" xyz="0 0 0.330"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.92" lower="-2.9670" upper="2.9670" velocity="7.9412"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_2_l" type="revolute">
    <parent link="link_1_s"/>
    <child link="link_2_l"/>
    <origin rpy="0 0 0" xyz="0.040 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="176.4" lower="-1.1344" upper="2.5307" velocity="6.7495"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_3_u" type="revolute">
    <parent link="link_2_l"/>
    <child link="link_3_u"/>
    <origin rpy="0 0 0" xyz="0 0 0.345"/>
    <axis xyz="0 -1 0"/>
    <limit effort="107.56" lower="-1.2217" upper="3.3161" velocity="9.0757"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_4_r" type="revolute">
    <parent link="link_3_u"/>
    <child link="link_4_r"/>
    <origin rpy="0 0 0" xyz="0.340 0 0.040"/>
    <axis xyz="-1 0 0"/>
    <limit effort="55.86" lower="-3.3161" upper="3.3161" velocity="9.5993"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_5_b" type="revolute">
    <parent link="link_4_r"/>
    <child link="link_5_b"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="32.68" lower="-2.3561" upper="2.3561" velocity="9.5993"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <joint name="joint_6_t" type="revolute">
    <parent link="link_5_b"/>
    <child link="link_6_t"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="-1 0 0"/>
    <limit effort="22.54" lower="-6.2944" upper="6.2944" velocity="17.4845"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- end of joint list -->
  
  <transmission name="transmission_h">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_0_s">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_h">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_s">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_1_s">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_s">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_l">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_2_l">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_l">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_u">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_3_u">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_u">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_r">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_4_r">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_r">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_b">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_5_b">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_b">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_t">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_6_t">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_t">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>

